### PR TITLE
Fix resolution check tests for mobile devices.

### PR DESCRIPTION
**Pull request by andresusanopinto**
When doing so also:
 - Refactor FrameChecker out of the test class and as a utility class than
   can be used to extract stats on what is being rendered into a video
   element.
 - Moved time to setup back to the test instead of having such specific
   code into call.js
 - Fix the resolution test, it was not working due to accessing non
   existent vars.

### CODE REVIEW

File: src/js/camresolutionstest.js [hunk 1]
```diff
@@ -220,21 +219,26 @@ CamResolutionsTest.prototype = {
       statsReport.avgSentFps = arrayAverage(googAvgFrameRateSent);
       statsReport.minSentFps = arrayMin(googAvgFrameRateSent);
       statsReport.maxSentFps = arrayMax(googAvgFrameRateSent);
-      statsReport.isMuted = this.isMuted;
-      statsReport.readyState = this.stream.getVideoTracks()[0].readyState;
-      statsReport.testedFrames = this.numFrames;
-      statsReport.blackFrames = this.numBlackFrames;
-      statsReport.frozenFrames = this.numFrozenFrames;
+      statsReport.isMuted = frameStats.isMuted;
```
Comment (KaptenJansson) [inline src/js/camresolutionstest.js]: frameStats.isMuted never gets assigned a value.

Comment (andresusanopinto) [inline src/js/camresolutionstest.js]: Right.... this was the variable I left in "this". Fixed.

**Comment by KaptenJansson** [comment]
LGTM

**Comment by KaptenJansson** [comment]
Pushed, served and tested on both desktop and mobile. One thing to note however, the camera API in lollipop is more sensitive causing chrome to hang. Will investigate this separately.

**Pull request merged**